File: cpp/0019删除链表的倒数第N个结点.cpp
#include"mylinklist.h"

using namespace std;


ListNode* removeNthFromEnd(ListNode*head,int n)
{
    ListNode *extra_head = new ListNode();
    extra_head->next = head;
    ListNode *slow = extra_head;
    ListNode *fast = head;
    
    while(n-->0)
        fast = fast->next;
    
    while(fast)
    {
        fast = fast->next;
        slow = slow->next;

    }
    slow->next = slow->next->next;
    return extra_head->next;
}

int main()
{
    vector<int> v = {1, 2, 3};

    Show(Create(v));
}

File: python/main/0019删除链表的倒数第N个结点.py
from linklist import *


def removeNthFromEnd(head: ListNode, n: int) -> ListNode:
    extra_head=ListNode(next=head)

    slow=extra_head
    fast=head

    while(n>0):
        fast=fast.next
        n-=1

    while(fast):
        fast=fast.next
        slow=slow.next

    slow.next=slow.next.next

    return extra_head.next


head=create([1,2])
show(removeNthFromEnd(head,2))
